### ISSUE TITLE
Combined naming bug with on-demand courses

### ISSUE THREAD

**Issue by Trailglider** [coursera-dl/coursera-dl#403]
I have tried the --combined-section-lectures-nums option with two on-demand courses, and for both I end up with several files beginning with "01_01". This is a big problem for me, as it means my phone will not list the videos in right order.

It seems the problem is week numbers are ignored and files are named by LL_VV_nameofvideo where LL=Lesson number and VV=Video number. If files could be prepended with week number the problem would be solved (named WW_LL_VV_nameofvideo where WW=Week number).

Further to this, in my opinion, the combined naming scheme should be the default - it adds very little length to the filename. I haven't been able to test with regular non-on-demand courses yet, so don't know if this is isolated to only on-demand courses or a general problem.

**Comment by victorwestmann** [comment]
So this is a feature request I believe.

**Comment by Trailglider** [comment]
Call it as you will. I was able to get regular courses to download and can confirm this only affects on-demand courses.

**Comment by rbrito** [comment]
I now know a little bit more (not much) about on-demand courses (since that's the future of coursera).

It seems that, contrary to what was the case for regular courses, with just 2 levels of names (the section/week name and then the video name), the on-demand courses have 3 levels of names (apparently, something along the lines of section/week name, subject, and then video name).

Just curious @Trailglider, are you using the  --combined-section-lectures-nums option due to the filesystem long names limitation or for any other reason?

What can you show us some kind of output that you get?

**Comment by jrao1** [comment]
Yeah, this is a problem for me too. I think it's needed because some devices such as iOS doesn't have the concept of folder, so we need to store all the videos in one folder, in order.

For an example, I'm download the course iot using --combined-section-lectures-nums option, it results in the following structures:
iot/01_what-is-the-internet-of-things-iot/
iot/01_what-is-the-internet-of-things-iot/01_get-started/
iot/01_what-is-the-internet-of-things-iot/01_get-started/01_01_prof-harris-introduction.mp4
iot/01_what-is-the-internet-of-things-iot/02_module-1/
iot/01_what-is-the-internet-of-things-iot/02_module-1/02_01_module-1-introduction.mp4
iot/01_what-is-the-internet-of-things-iot/02_module-1/02_02_prof-harris-takes-you-on-a-tour-of-his-lab.mp4
...
iot/02_embedded-systems/
iot/02_embedded-systems/01_module-2/
iot/02_embedded-systems/01_module-2/01_01_module-2-introduction.mp4
...

When you put all the mp4 files in one folder, 01_01_prof-harris-introduction.mp4 will be next to 01_01_module-2-introduction.mp4, which is bad.
